Jenkinsfile

```diff
@@ -28,292 +28,319 @@
+                            dir('Palisade-services-deploy') {
+                                if (IS_PR == "true" || FEATURE_BRANCH == "false") {
+                                    git branch: GIT_BRANCH_NAME, url: 'https://github.com/gchq/Palisade-services.git'
+                                    sh 'palisade-login'
+                                    //now extract the public IP addresses that this will be open on
+                                    sh 'extract-addresses'
+                                    // Push containers to the registry so they are available to helm
+                                    if (FEATURE_BRANCH == "true") {
+                                        if (sh(script: "namespace-create ${GIT_BRANCH_NAME_LOWER}", returnStatus: true) == 0) {
+                                            sh 'echo namespace create succeeded'
+                                        } else {
+                                            error("Could not create namespace")
+                                        }
+                                    }
+                                    sh "mvn -s ${MAVEN_SETTINGS} -Dmaven.test.skip=true -P pi -D revision=${SERVICES_REVISION} -D common.revision=${COMMON_REVISION} -D readers.revision=${READERS_REVISION} deploy"
+                                    //deploy application to the cluster
+                                    if (sh(script: "helm upgrade --install palisade . " +
+                                            "--set global.hosting=aws  " +
+                                            "--set traefik.install=false,dashboard.install=false " +
+                                            "--set global.repository=${ECR_REGISTRY} " +
+                                            "--set global.hostname=${EGRESS_ELB} " +
+                                            "--set global.deployment=example " +
+                                            "--set global.persistence.dataStores.palisade-data-store.aws.volumeHandle=${VOLUME_HANDLE_DATA_STORE} " +
+                                            "--set global.persistence.classpathJars.aws.volumeHandle=${VOLUME_HANDLE_CLASSPATH_JARS} " +
+                                            "--set global.redisClusterEnabled=true " +
+                                            "--set global.redis.install=false " +
+                                            "--set global.redis-cluster.install=true " +
+                                            "--namespace ${HELM_DEPLOY_NAMESPACE}", returnStatus: true) == 0) {
+                                        echo("successfully deployed")
+                                    } else {
+                                        error("Build failed because of failed maven deploy")
+                                    }
+                                    if (FEATURE_BRANCH == "true") {
```

Comment:
I'm kind of thinking that this should be IS_PR == "true"? Am I right?

Reply:
So either should work as it only gets into that code block if it is a PR or not a feature branch.

Reply:
I think this is correct. The `IS_PR` flag is already true due to the initial if statement in this stage and we only want to create the namespace and delete the namespace if `FEATURE_BRANCH` is true.